flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -27,9 +27,30 @@ void TableColumn(TableCreationContext context) :
         UniqueKey(context.uniqueKeysList)
     |
         ComputedColumn(context)
+    |
+        Watermark(context)
     )
 }
 
+void Watermark(TableCreationContext context) :
+{
+    SqlIdentifier columnName;
+    SqlParserPos pos;
+    SqlNode watermarkStrategy;
+}
+{
+    <WATERMARK> <FOR>
```

Comment:
Could we also remove the reserved keywords from the old Blink design? I think there are still commented out test cases and keywords in the parser.

Reply:
Good point.